h/groups/test/views_test.py

```diff
@@ -349,6 +349,125 @@ def test_read_if_already_a_member_returns_response(Group, renderers):
     assert views.read(request) == mock.sentinel.response
 
 
+@read_fixtures
+def test_read_calls_search(Group, search, renderers):
+    """It should call search() to get the annotations."""
+    request = _mock_request(matchdict=_matchdict())
+    g = Group.get_by_pubid.return_value = mock.Mock(slug=mock.sentinel.slug)
+    user = request.authenticated_user = mock.Mock()
+    user.groups = [g]  # The user is a member of the group.
+    renderers.render_to_response.return_value = mock.sentinel.response
+
+    views.read(request)
+
+    search.search.assert_called_once_with(
+            request, private=False, params={"group": g.pubid, "limit": 1000})
+
+
+@read_fixtures
+def test_read_calls_normalize(Group, search, renderers, uri):
+    """It shold call normalize() with each of the URIs from search()."""
```

Comment:
I'm not really sure about this test. Again, what I think you actually want to test is that if `search` returns N annotations whose URIs differ only in normalisation, you only get one back, no? There are many ways of doing this, some of which don't involve calling `uri.normalize` (e.g. partition on `target.source` rather than `uri.normalize(annotation.uri)`).